com.unity.render-pipelines.high-definition/Runtime/Lighting/LightCookieManager.cs

```diff
@@ -223,12 +223,55 @@ public void LayoutIfNeeded()
             }
         }
 
+        public Vector4 Fetch2DCookie(CommandBuffer cmd, Texture cookie, Texture ies)
+        {
+            int width  = (int)Mathf.Max(cookie.width, ies.height);
+            int height = (int)Mathf.Max(cookie.width, ies.height);
+
+            if (width < k_MinCookieSize || height < k_MinCookieSize)
+                return Vector4.zero;
+
+            if (!m_CookieAtlas.IsCached(out var scaleBias, m_CookieAtlas.GetTextureID(cookie, ies)) && !m_NoMoreSpace)
+                Debug.LogError($"2D Light cookie texture {cookie} can't be fetched without having reserved. You can try to increase the cookie atlas resolution in the HDRP settings.");
+
+            if (m_CookieAtlas.NeedsUpdate(cookie, ies, false))
+            {
+                RTHandle multiplied = RTHandles.Alloc(  width, height,
+                                                        colorFormat: cookieFormat,
+                                                        enableRandomWrite: true);
+                RTHandle tex0;
+                RTHandle tex1;
+                if (cookie.width*cookie.height > ies.width*ies.height)
+                {
+                    tex0 = RTHandles.Alloc(cookie);
+                    tex1 = RTHandles.Alloc(ies);
+                }
+                else
+                {
+                    tex0 = RTHandles.Alloc(ies);
+                    tex1 = RTHandles.Alloc(cookie);
+                }
+                cmd.SetRenderTarget(multiplied);
+                cmd.ClearRenderTarget(false, true, Color.black);
+                GPUArithmetic.ComputeOperation(multiplied, tex0, cmd, GPUArithmetic.Operation.Add);
+                GPUArithmetic.ComputeOperation(multiplied, tex1, cmd, GPUArithmetic.Operation.Mult);
+
+                m_CookieAtlas.BlitTexture(cmd, scaleBias, multiplied, new Vector4(1, 1, 0, 0), blitMips: false, overrideInstanceID: m_CookieAtlas.GetTextureID(cookie, ies));
+
+                RTHandlesDeleter.ScheduleRelease(tex0);
```

Comment:
Do it mean we reuse memory each frame if the texture is animated, or does it mean we will allocate the memory 3 times before freeing it?

Reply:
We talk about this comment?
```
// TODO: we don't need to allocate two temp RT, we can use the atlas as temp render texture
// it will avoid additional copy of the whole mip chain into the atlas.
```
This comment is copy pasted from the previous function:
Texture FilterAreaLightTexture(CommandBuffer cmd, Texture source)Texture FilterAreaLightTexture(CommandBuffer cmd, Texture source)

Where the computation & the mip-filling is done with Draw/Blit. In my side I do that with ComputeShader & CopyTexture functions.